File: dotnet/src/Pactman.NonprofitCheckPlus/Exceptions/PactmanRateLimitException.cs
namespace Pactman.NonprofitCheckPlus.Exceptions
{
    /// <summary>
    /// HTTP 429. <see cref="PactmanApiException.RetryAfter"/> carries the server's
    /// <c>Retry-After</c> when it sent one.
    /// </summary>
    public sealed class PactmanRateLimitException : PactmanApiException
    {
        /// <summary>Initializes the exception from response metadata.</summary>
        /// <param name="message">Human-readable description of the failure.</param>
        /// <param name="init">The metadata the transport collected from the response.</param>
        public PactmanRateLimitException(string message, ApiErrorInit init)
            : base(message, init, ErrorCategory.RateLimit)
        {
        }
    }
}

File: java/sdk/src/main/java/org/pactman/nonprofitcheckplus/exceptions/PactmanRateLimitException.java
package org.pactman.nonprofitcheckplus.exceptions;

/** HTTP 429. {@link #retryAfterSeconds()} carries the server's {@code Retry-After} when it sent one. */
public class PactmanRateLimitException extends PactmanApiException {

    private static final long serialVersionUID = 1L;

    /**
     * Creates the exception.
     *
     * @param message the message to surface.
     * @param init    the response metadata.
     */
    public PactmanRateLimitException(String message, ApiErrorInit init) {
        super(message, init, ErrorCategory.RATE_LIMIT);
    }
}
